modules/apps/commerce/commerce-product-asset-categories-navigation-web/src/main/java/com/liferay/commerce/product/asset/categories/navigation/web/internal/display/context/CPAssetCategoriesNavigationDisplayContext.java

```diff
@@ -253,9 +287,27 @@ public String getFriendlyURL(long categoryId, ThemeDisplay themeDisplay)
 			friendlyURLEntry.getUrlTitle(languageId);
 	}
 
-	public String getRootAssetCategoryId() {
-		return _cpAssetCategoriesNavigationPortletInstanceConfiguration.
-			rootAssetCategoryId();
+	public String getRootAssetCategoryExternalReferenceCode()
+		throws PortalException {
+
+		AssetCategory assetCategory = null;
+
+		String rootAssetCategoryExternalReferenceCode =
+			_cpAssetCategoriesNavigationPortletInstanceConfiguration.
+				rootAssetCategoryExternalReferenceCode();
+
+		if (Validator.isNotNull(rootAssetCategoryExternalReferenceCode)) {
+			assetCategory =
+				_assetCategoryService.fetchCategoryByExternalReferenceCode(
+					rootAssetCategoryExternalReferenceCode,
+					_themeDisplay.getCompanyGroupId());
+		}
+
+		if (assetCategory != null) {
+			return String.valueOf(assetCategory.getCategoryId());
```

Comment:
Why is the method called `getRootAssetCategoryExternalReferenceCode` and we are returning the ID?